File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/pdf/ImageTagProcessor.java
package org.javamaster.invocationlab.admin.pdf;

import com.itextpdf.text.Chunk;
import com.itextpdf.text.Element;
import com.itextpdf.text.Image;
import com.itextpdf.text.pdf.codec.Base64;
import com.itextpdf.tool.xml.NoCustomContextException;
import com.itextpdf.tool.xml.Tag;
import com.itextpdf.tool.xml.WorkerContext;
import com.itextpdf.tool.xml.exceptions.RuntimeWorkerException;
import com.itextpdf.tool.xml.html.HTML;
import com.itextpdf.tool.xml.pipeline.html.HtmlPipelineContext;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class ImageTagProcessor extends com.itextpdf.tool.xml.html.Image {


    @Override
    public List<Element> end(final WorkerContext ctx, Tag tag, List<Element> currentContent) {
        final Map<String, String> attributes = tag.getAttributes();
        String src = attributes.get(HTML.Attribute.SRC);
        List<Element> elements = new ArrayList<>(1);
        if (src == null || src.isEmpty()) {
            return elements;
        }
        Image img = null;
        if (src.startsWith("data:image/")) {
            final String base64Data = src.substring(src.indexOf(",") + 1);
            try {
                img = Image.getInstance(Base64.decode(base64Data));
            } catch (Exception e) {
                throw new RuntimeWorkerException(e);
            }
            try {
                HtmlPipelineContext htmlPipelineContext = getHtmlPipelineContext(ctx);
                Chunk chunk = new Chunk((Image) getCssAppliers().apply(img, tag, htmlPipelineContext), -50, 0, true);
                elements.add(getCssAppliers().apply(chunk, tag, htmlPipelineContext));
            } catch (NoCustomContextException e) {
                throw new RuntimeWorkerException(e);
            }
        }
        if (img == null) {
            elements = super.end(ctx, tag, currentContent);
        }
        return elements;
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/pdf/ImageTagProcessor.kt
package org.javamaster.invocationlab.admin.pdf

import com.itextpdf.text.Chunk
import com.itextpdf.text.Element
import com.itextpdf.text.pdf.codec.Base64
import com.itextpdf.tool.xml.NoCustomContextException
import com.itextpdf.tool.xml.Tag
import com.itextpdf.tool.xml.WorkerContext
import com.itextpdf.tool.xml.exceptions.RuntimeWorkerException
import com.itextpdf.tool.xml.html.HTML
import com.itextpdf.tool.xml.html.Image

class ImageTagProcessor : Image() {
    override fun end(ctx: WorkerContext, tag: Tag, currentContent: List<Element>): List<Element> {
        val attributes = tag.attributes
        val src = attributes[HTML.Attribute.SRC]
        var elements: MutableList<Element> = ArrayList(1)
        if (src.isNullOrEmpty()) {
            return elements
        }
        var img: com.itextpdf.text.Image? = null
        if (src.startsWith("data:image/")) {
            val base64Data = src.substring(src.indexOf(",") + 1)
            try {
                img = com.itextpdf.text.Image.getInstance(Base64.decode(base64Data))
            } catch (e: Exception) {
                throw RuntimeWorkerException(e)
            }
            try {
                val htmlPipelineContext = getHtmlPipelineContext(ctx)
                val chunk =
                    Chunk(cssAppliers.apply(img, tag, htmlPipelineContext) as com.itextpdf.text.Image, -50f, 0f, true)
                elements.add(cssAppliers.apply(chunk, tag, htmlPipelineContext))
            } catch (e: NoCustomContextException) {
                throw RuntimeWorkerException(e)
            }
        }
        if (img == null) {
            elements = super.end(ctx, tag, currentContent)
        }
        return elements
    }
}